src/streaming.ts

```diff
@@ -1,11 +1,14 @@
-import EventSource from '@harnessio/eventsource'
 import EventEmitter from 'events';
 import { AxiosPromise } from 'axios';
 import { ClientApi, FeatureConfig, Segment } from './openapi';
 import { StreamEvent, Options, StreamMsg } from './types';
 import { Repository } from './repository';
 import { ConsoleLog } from './log';
 
+import * as https from 'https';
+import * as http from 'http';
+import { RequestOptions } from 'https';
```

Comment:
I think you can simplify this as just...
```suggestion
import https, { RequestOptions } from 'https';
import http from 'http';
```

It's also good practice to import things that are from the platform first (in this case before `EventEmitter`), but deffo not required